cloudgenix_config/pull.py

```diff
@@ -851,7 +854,10 @@ def _pull_config_for_single_site(site_name_id):
             if_type = interface.get('type')
             if not FORCE_PARENTS and interface_id in parent_id_list:
                 # interface is a parent, skip
-                continue
+                # Pull interface config for bypasspair and virtual interface as it can have subif/pppoe/servicelink configs
+                # And its mandatory that parent gets created first
+                if if_type not in ('virtual_interface', 'bypasspair'):
```

Comment:
The reason for pulling the parent VI/bypasspair is because for a subif/pppoe/servicelink to be created, the parent must first be created and must be available on controller. Now if i am trying to create VI->subif, and if VI config is not present in yml, then during subif creation an error will be thrown saying parent not present. Thats the reason for pulling VI and bypass if its a parent...

Like you said subif and all can have 'port' type as parent. But in that case, the port will anyway be present on the controller and its not mandatory for it to be available in the yml..

I hope that makes sense